### ISSUE TITLE
Stopping builds triggered by a parent build

### ISSUE THREAD

**Issue by rpurdie** [buildbot/buildbot#3988]
If I have a builder which triggers a set of other builders, I can stop the parent build but I can't seem to find a way of stopping all the 'child' triggered builds. It would be really useful to be able to have a way to do this. I couldn't find a general "stop all running builds" button either, or any documentation on doing that.

**Comment by tardyp** [comment]
Hi @rpurdie ,
Newer versions (aka nine) of Buildbot do stop child builds when you stop a parent build, that is waiting for its childs.
I think that oe buildbot is still on an ancient version of buildbot "eight", which does not have this feature.

**Comment by rpurdie** [comment]
Hi, Thanks for the reply! You're right, I'm referring to the OE/Yocto Project infrastructure however we're in the process of testing an updated setup using buildbot 1.0.0. Should that version have this feature and is there any config I need to set to ensure that behaviour? Cheers, Richard

**Comment by tardyp** [comment]
Nothing specific to be configured. I don't see a usecase to stop the parent build without wanting to stop the childs...

**Comment by rpurdie** [comment]
Am I right in thinking the build needs to be waiting on the triggered builds in order to get that behaviour? That may be where we're going wrong as we wait on some but not all.

**Comment by tardyp** [comment]
Right, that only works for a trigger step that is still alive via the
interrupt method

See implementation detail:
https://github.com/buildbot/buildbot/blob/master/master/buildbot/steps/trigger.py#L108

If you  want a stop everything script, you can just send that control REST
api to cancel all buildrequests that are still active.

Pierre

**Comment by rpurdie** [comment]
I hadn't realised that it would/could only cancel things which it was waiting on although it makes sense now I understand the code. I've changed the way we structured the builds so it now does what we really need, when we stop it, it stops the set in question. That should be much more convenient than stopping all builds for us. It might be useful to mention that detail in the trigger docs somewhere but I have my situation working now, thanks!
